scripts/mark_eats/scales_demo.lua

```diff
@@ -0,0 +1,161 @@
+-- Scales Demo
+--
+-- Example scale generation
+-- using MusicUtil.
+-- 
+-- ENC1 : Tempo
+-- ENC2 : Root note
+-- ENC3 : Scale
+-- KEY2 : Play/Pause
+-- KEY3 : Reset step
+-- 
+
+local MusicUtil = require "mark_eats/musicutil"
+
+local root_note = 48
+local scale_type = 1
+local scale_notes = {}
+local scale_note_names = {}
+local scale_len = 0
+local SCALES_LEN = #MusicUtil.SCALES
+
+local tempo = 90
+local step = 1
+local step_increment = 1
+local step_metro
+
+engine.name = "PolyPerc"
+
+
+local function init_scale()
+  scale_notes = MusicUtil.generate_scale(root_note, scale_type, 1)
+  scale_note_names = MusicUtil.note_nums_to_names(scale_notes)
+  scale_len = #scale_notes
+end
+
+local function advance_step()
+  if step > scale_len then
+    step = scale_len
+    step_increment = -1
+  else
+    step = step + step_increment
+    if step == 1 or step == scale_len then
+      step_increment = step_increment * -1
+    end
+  end
+  engine.hz(MusicUtil.note_num_to_freq(scale_notes[step]))
```

Comment:
aside: it'd be interesting to consider adding a `note` function to "PolyPerc" (and related engines).  i bet lots of folks would find that handy!

Reply:
Yes I think having proper consistency between the engines would be a very good idea. At least making everything respond to `freq` and `gate` which seems standard in SC.